### ISSUE TITLE
Discussion of badge draft

### ISSUE THREAD

**Issue by rjleveque** [uwescience/reproducible#3]
For discussion of the draft found on the wiki page [[DRAFT] Open Science and Reproducible Badges](https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges).

Thanks @sr320 for getting this started.

Question: Do you see this as a badge for individual researchers or for a research group / lab?  If the latter, it may be hard to figure out how to assign points without more guidance.

**Comment by sr320** [comment]
I was thinking at the individual level.

**Comment by benmarwick** [comment]
I agree that the individual level would be the simplest to start with.

**Comment by dacb** [comment]
Some of the items in cat. 1 appear to be specific to a publication.  Is that the intention?

**Comment by benmarwick** [comment]
I've had a go at filling in some of the gaps following the discussion in the meeting today. Please have a look to see if I've forgotten or misremembered something.

**Comment by arokem** [comment]
I like the Mozilla platform. It's obviously been thought through. For example, the fact that the badge links back to information about how the badge was issued and verifies it makes it much more meaningful than if people just testify about themselves that they are "reproducible". Even if they do their own self-assessment, it's still more meaningful than leaving it up to a vague interpretation.

Getting Mozilla's approval of our badging scheme is a badge of approval for that badging scheme. A meta-badge! :-)

**Comment by benmarwick** [comment]
@arokem are you interested to look further into the Mozilla platform and see what might be involved in setting up our own BadgeKit?

**Comment by dacb** [comment]
@arokem That would be fantastic!

**Comment by arokem** [comment]
I will look into it and post some impressions on this issue.

**Comment by arokem** [comment]
## Mozilla Open Badge Infrastructure (MOBI)

See [here](https://wiki.mozilla.org/Badges)

Provides a conceptual and technical framework for granting individuals an
online record of specific achievements, linking to the set of interactions
that
led to the issuing of the badge, and the work completed to get it.

=> MOBI badges are granted to individuals, not projects.

We could potentially grant a badge to an individual for having a project
with a certain
standard, and then 'stack' badges for having several projects related to a
specific individual. Different badges can be granted for learning about
something (e.g. going through an online training module), or for passing
some
(self?) audit.

=> MOBI supports verification of the badges with respect to an issuing
organization: A
centralized authority, or a community, which certifies the achievement.

One example from UC Davis can be found [here](
http://www.reconnectlearning.org/wp-content/uploads/2014/01/UC-Davis_case_study_final.pdf
)

Specifically, MOBI provides a set of free software tools, and an open
technical
standard that can be used by a badge-granting authority to facilitate the
administering and verification of badges. These tools facilitate the
setting up
and administering a web app for badging, and a data-base of badges granted.
For
details see [this page](
https://github.com/mozilla/openbadges-badgekit/wiki/BadgeKit-Self-Hosting-Guide)
and [this page](
https://github.com/mozilla/badgekit-api/wiki/Using-BadgeKit-API)

Advantages: standard, thought-through way of doing this. Plugs into already
established mechanisms for displaying and collecting badges (e.g. Mozilla's
'backpack').

Disadvantages: requires technical work to set up and administer - there
might be easier ways to do this (?). Will
require setting clear and uniform (self?) evaluation criteria, which may
not be an achievable goal at this time.

**Comment by benmarwick** [comment]
@arokem would you like to try and set up a local instance of this infrastructure?

**Comment by rjleveque** [comment]
It would be great to try this out.

**Comment by arokem** [comment]
Not unless we first collectively decide that we're actually going to use
it, and more importantly decide *how* we're going to use it. If you want to
get a sense for what it would look and feel like, you can check out the
mobi site.

**Comment by rjleveque** [comment]
I think some of the things listed in Category 1 in the Draft should be moved to Category 2.  As it stands, the Category 2 list only concerns data, not code.  Things like "have a GitHub account", "Code has an open license", and "Code has persistent URL or DOI" seem to belong under "open" rather than "reproducible"? 

Another question: it says the badge is good for one academic year. So in adding up points, e.g. 4 points for a publication with an associated repository or 1 point for each publication on the arXiv, does one only count publications within the last year?

**Comment by sr320** [comment]
@rjleveque I agree with you on the categories. 

I think we can also come up with additional criteria. 

Regarding the one academic year, personally I think this makes it too complicated. I think it should be you get a badge. Should be designed where there is room for growth (ie not many should be able to get Gold on the onset) so persons would re-assess to go up a level.

**Comment by benmarwick** [comment]
I've made some edits to incorporate those suggestions about classification of criteria and removing the one year limit. 

I just came across this related project: [http://www.artifact-eval.org/about.html](http://www.artifact-eval.org/about.html), which gives out these badges:

![clipboard01](https://cloud.githubusercontent.com/assets/1262179/7332006/f6b41d7c-eadf-11e4-9c61-c64454610ab8.png)

**Comment by rjleveque** [comment]
Thanks @benmarwick!  I mostly like the changes.  Regarding the items in Category 2 such as "Code has a persistent URL" or "Data in UW ResearchWorks", is this meant to give 1 point for each such item, or 1 point total if you have at least one such item?

I didn't know about Publon.  Should we also give a point for signing up for things like ORCID as a way to encourage it?

**Comment by sr320** [comment]
I was thinking 1 point per item. 
Yes I agree that giving points for services that we want to encourage or highlight like ORCID would be good.

**Comment by kallisons** [comment]
- [ ] points for entering data in the catalog
- [ ] catalog of badge items (that don't fit in the existing paper/dataset catalog)

**Comment by benmarwick** [comment]
This is a nice system for describing the openness of data that we might want to make mention of: [5 ★ Open Data](http://5stardata.info/)

**Comment by sr320** [comment]
Some examples of simple badges

![starter](https://img.shields.io/badge/ROS%20Badge--grey.svg)

![silver](https://img.shields.io/badge/ROS%20Badge-36-lightgrey.svg)

![gold](https://img.shields.io/badge/ROS%20Badge-75-yellow.svg)

code

```
![starter](https://img.shields.io/badge/ROS%20Badge--grey.svg)

![silver](https://img.shields.io/badge/ROS%20Badge-36-lightgrey.svg)

![gold](https://img.shields.io/badge/ROS%20Badge-75-yellow.svg)
```

Would want them to go back to webpage (that eventually links out to list persons with badges & their details)


[![starter](https://img.shields.io/badge/ROS%20Badge--grey.svg)](https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges)

[![silver](https://img.shields.io/badge/ROS%20Badge-36-lightgrey.svg)](https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges)

[![gold](https://img.shields.io/badge/ROS%20Badge-75-yellow.svg)](https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges)


code

```
[![starter](https://img.shields.io/badge/ROS%20Badge--grey.svg)](https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges)

[![silver](https://img.shields.io/badge/ROS%20Badge-36-lightgrey.svg)](https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges)

[![gold](https://img.shields.io/badge/ROS%20Badge-75-yellow.svg)](https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges)
```

corresponding html that someone code embed 

<a href="https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges"> <img src="https://img.shields.io/badge/ROS%20Badge-75-yellow.svg" alt="ROS Badge"></a>




code
```
<a href="https://github.com/uwescience/reproducible/wiki/%5BDRAFT%5D-Open-Science-and-Reproducible-Badges"> <img src="https://img.shields.io/badge/ROS%20Badge-75-yellow.svg" alt="ROS Badge"></a>
```

**Comment by sr320** [comment]
# Method for Calculating Scores from Google Sheets

Raw data looks something like
<img src="http://eagle.fish.washington.edu/cnidarian/skitch/-_Google_Sheets_1B20DC68.png" alt="-_Google_Sheets_1B20DC68.png"/>

I created a second sheetand simply transferred header with `=Sheet1!A1` in cell `A1` etc along row 1.

For the first data cell _Publications with Repo_ I counted the number of lines with `(LEN(Sheet1!B2) - LEN(SUBSTITUTE(Sheet1!B2, CHAR(10), "")) + 1)` and multiplied by 4 (points per publication) to get `=(LEN(Sheet1!B2) - LEN(SUBSTITUTE(Sheet1!B2, CHAR(10), "")) + 1)*4`

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/-_Google_Sheets_1B20DD3F.png" alt="-_Google_Sheets_1B20DD3F.png"/>

If it was a entry where a single point value was given for any entry I used `=if(Sheet1!D2<>"",1,"")`.  '1' would be changed based on point value and nothing would show up if nothing entered.

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/-_Google_Sheets_1B20DD89.png" alt="-_Google_Sheets_1B20DD89.png"/>

Finally I just summed up the rows!
<img src="http://eagle.fish.washington.edu/cnidarian/skitch/-_Google_Sheets_1B20DDEF.png" alt="-_Google_Sheets_1B20DDEF.png"/>

Things that need attention is noting on form to leave blank if nothing to add (that is why my score is inflated). 


---

The sheet I used to test this approach out is [here](https://docs.google.com/spreadsheets/d/1Kjjjx6Yetz8MyRSkkHaCqNm6ZnXK6sqkqCyHQmItoPI/edit?usp=sharing)

**Comment by benmarwick** [comment]
Some comments on entering data:
- Submit your code for review: Question not that clear,
- Have the substantive results of your publication reproduced by someone else using your code and data: Question not clear
- Code has one or more tests: not clear what this means
- One or more sets of raw data are published as soon as generated: tough question - maybe edit
- One or more sets of raw data are in open formats (eg. CSV, txt, etc.) : maybe redundant?
- What percentage of your publications from the last two years are Open Access?: not easy

**Comment by benmarwick** [comment]
Inspired by Steven's summaries, I made a simple shiny app so can see the scores and details of each respondent. The app pulls data from the google form (and scrapes the wiki page for the scoring), tallies the score and tells you what badge you earned. It also shows the score per criterion and the detailed responses. 

There's a drop-down menu that is populated by the names of the respondents, when a different name is chosen, the other tables auto-update.

![uw escience reproducibility badge survey responses](https://cloud.githubusercontent.com/assets/1262179/7996165/6e3b8724-0ad2-11e5-9c06-d2dbfeda9fc4.png)

There's a minor mismatch at the bottom of the bottom table. 

Here's the code for my shiny app:

ui.R
```
shinyUI(fluidPage(
  titlePanel("UW eScience Reproducibility Badge Survey Responses"),
  
  sidebarLayout(

    sidebarPanel(
      selectInput("nameInput", "Please choose a person to see their responses:", unlist(tmp[2]))
    ),
    
    
    
    mainPanel("Reproducibility Badge Score",
              tableOutput("my_table1"),
              tableOutput("my_table"),
              tableOutput("my_table2")
    )
  )
))
```

server.R
```



######################

shinyServer(function(input, output) {
  

  
# output table of score per criterion
  
  output$my_table <- renderTable({ 

    numConv <- reactive({as.numeric( which(unlist(tmp[2]) ==  input$nameInput ))})
    n <- numConv()
    # get all responses from person n
    person <- lapply(tmp, function(i) unlist(i[n]))
    person_ <- person[-c(1:2)]
    scr_c <- scr_c_f(person_, n)
    # score for each criterion
    scr_df <- data.frame(criteria = scores$Criterion, 
                         score = scr_c[1:24])
    scr_df
  })

# output table of person's name and total score
  output$my_table1 <- renderTable({  
    numConv <- reactive({as.numeric( which(unlist(tmp[2]) ==  input$nameInput ))})
    n <- numConv()
    person <- lapply(tmp, function(i) unlist(i[n]))
    person_ <- person[-c(1:2)]

    scr_c <- scr_c_f(person_, n)
    scr_df <- data.frame(criteria = scores$Criterion, 
                         score = scr_c[1:24])
    sums <- sum(scr_c[1:24], na.rm = TRUE)
    my_table1 <- data.frame(person = input$nameInput,
                            total_score = sums,
                             badge = ifelse(sums >= 50, "Gold", 
                                        ifelse(sums %in% 25:49, "Silver",   "Bronze")))
    my_table1
  })
  
# output table of specific responses to each criterion
  output$my_table2 <- renderTable({ 
    # split all the reponses by line breaks

    
    numConv <- reactive({as.numeric( which(unlist(tmp[2]) ==  input$nameInput ))})
    n <- numConv()
    # get response detail for specific person
    person <- lapply(tmp, function(i) unlist(i[n]))
    # make into a table
    data.frame(prompt = c("date", "name", scores$Criterion[1:24]),
      responses = sapply(person, function(i) paste0(i, collapse = ", "))[1:26] )
    
    
  })
  
  
})

```

**Comment by sr320** [comment]
An update on our progress ----   http://tinyurl.com/nvqza2g
A single webpage that provides access to survey, results, and badges

**Comment by arokem** [comment]
Very cool!

In the meanwhile, we've also managed to set up an instance of the Mozilla
Badgekit here:

http://45.55.180.20:3000

As anticipated above, it was a fun (and educational!) couple of days of
hacking to get this going. With a bit more elbow-grease, this could
possibly provide the backend to the system that Ben and Steven put together
to link it up to the Open Badge system. My feeling at the moment is that
this would be worthwhile if we anticipated creating other badges, and
several different front-ends for different kinds of certification on top of
this.

But give it a go around with this, and see what you think. Anyone with a `@
uw.edu` address should be able to log into our instance.

**Comment by arokem** [comment]
Just to let you know that I will be turning of the service on that machine in a couple of days, so if you haven't had the chance to try it out, please do so before Monday evening.

**Comment by benmarwick** [comment]
I haven't been able to login at all, but have you spoke with Bill Howe about a potential future for badge kit in escience?

**Comment by arokem** [comment]
What happened when you tried logging in? Was it just an authentication
issue, or was the badge-kit webpage not there at all?

I have not spoken to Bill about this yet.

**Comment by benmarwick** [comment]
I can see the page ok, I get a message about an incorrect password, I"m attempting to log in with my UWNetID credentials

**Comment by arokem** [comment]
I see - I think that you need to create a password for your uw email via
Persona <https://www.mozilla.org/en-US/persona/>. I don't remember what
exactly we did to get that going.

**Comment by rjleveque** [comment]
Some comments:

( `form` refers to http://goo.gl/forms/jm08DOJ2EI,  `app` refers to https://benmarwick.shinyapps.io/ros_badge/)

- I entered name and affiliation on two separate lines in the `form` and now there's an entry in the `app` for "Applied Mathematics, UW" with 0 points.  We should just allow one input line for this.

- The form doesn't ask for an ORC ID, but I still got credit for it on the `app`.

- I agree with @benmarwick's comments from June 4 about some questions not being clear.

- For some categories (e.g. code review) it's not clear if work on software development, e.g. on https://github.com/clawpack in my case, should be counted or whether we only mean to count code related to specific publications.

- Should there be a maximum number of points for some items, e.g. some people might have 50 preprints on the arXiv but have done very little to make code available, should this be gold?

**Comment by rjleveque** [comment]
Ideas from some discussions the past few days:

- Anyone interested in working more on these badges is welcome to join the working group on Tuesday Sept 22 at 1:30pm in the Data Science Studio for discussion and modification of the form.
- It would be great if we could "finalize" this enough to first test with some others from eScience, and then test more broadly at the Moore/Sloan DSE summit October 5-7.  (Sarah has reserved a breakout session time for this.)
- For this testing, it would be good to create another evaluation form where people could give feedback as they are going through the badges form.  Questions to ask on this include:
  - better wording of questions? 
  - are we asking about the right metrics?
  - usability of form?
  - other suggestions?

**Comment by sr320** [comment]
If it is helpful in terms of "finalizing" or getting everything ready to distribute- we also created a quick front-end with explanations, links etc during the sprint. 

Link: http://htmlpreview.github.io/?https://github.com/sr320/tmp-badge/blob/master/rros-badge-web.html


<img src="http://eagle.fish.washington.edu/cnidarian/skitch/README_1BA1FDD4.png" alt="README_1BA1FDD4.png"/>


---

In terms of another form, based on our effort just to get folks to fill out the form we made- it might be better to just have something very simple (page above just started issue).  I think most of the improvement will come from just getting people to fill in info and see what pops up in back end. We caught a lot during sprint with the 2-3 persons that filled it out.
